app/controllers/vacancies_controller.rb

```diff
@@ -10,13 +12,9 @@ def new
 
   def create
     @vacancy = Vacancy.new(vacancy_params)
-
-    if @vacancy.save
-      flash[:success] = t(".success")
-      redirect_to @vacancy
-    else
-      render :new
-    end
+    Vacancy::Publish.new(@vacancy).publish
```

Comment:
I prefer `Vacancy::Publish.new.publish(@vacancy)` I think it's more readable, i mean you publish a vacancy and it is what the code says, I like to use constructors to pass dependencies to the class or configurations. What do you think?